docs/advanced/server-side-rendering.md

```diff
@@ -177,254 +177,156 @@ we'll have to import from. `preact-ssr-prepass`.
-During the prepass of your component tree `next-urql` can't know how these functions will alter the props passed to your page component. This injection
-could change the `variables` used in your `useQuery`. This will lead to error being thrown during the subsequent `toString` pass, which isn't supported in React 16.
-
-### SSR with { ssr: true }
-
-The `withUrqlClient` only wraps our component tree with the context provider by default.
-To enable SSR, the easiest way is specifying the `{ ssr: true }` option as a second
-argument to `withUrqlClient`:
-
-```js
-import { dedupExchange, cacheExchange, fetchExchange } from '@urql/core';
-
-import { withUrqlClient } from 'next-urql';
+const { getClient } = registerUrql(makeClient);
 
-export default withUrqlClient(
-  ssrExchange => ({
-    url: 'http://localhost:3000/graphql',
-    exchanges: [dedupExchange, cacheExchange, ssrExchange, fetchExchange],
-  }),
-  { ssr: true } // Enables server-side rendering using `getInitialProps`
-)(Index);
+export default async function Home() {
+  const result = await getClient().query(PokemonsQuery, {});
+  return (
+    <main>
+      <h1>This is rendered as part of an RSC</h1>
+      <ul>
+        {result.data.pokemons.map((x: any) => (
+          <li key={x.id}>{x.name}</li>
+        ))}
+      </ul>
+    </main>
+  );
+}
 ```
 
-Be aware that wrapping the `_app` component using `withUrqlClient` with the `{ ssr: true }`
-option disables Next's ["Automatic Static
-Optimization"](https://nextjs.org/docs/advanced-features/automatic-static-optimization) for
-**all our pages**. It is thus preferred to enable server-side rendering on a per-page basis.
+When we aren't leveraging server components we will import the things we will
+need to do a bit more setup, we go to the `client` component's layout file and
+structure it as the following.
 
-### SSR with getStaticProps or getServerSideProps
+```tsx
+// app/client/layout.tsx
+'use client';
 
-Enabling server-side rendering using `getStaticProps` and `getServerSideProps` is a little
-more involved, but has two major benefits:
+import { UrqlProvider, ssrExchange, cacheExchange, fetchExchange, createClient } from '@urql/next';
 
-1. allows **direct schema execution** for performance optimisation
-2. allows performing extra operations in those functions
-
-To make the functions work with the `withUrqlClient` wrapper, return the `urqlState` prop
-with the extracted data from the `ssrExchange`:
-
-```js
-import { withUrqlClient, initUrqlClient } from 'next-urql';
-import { ssrExchange, dedupExchange, cacheExchange, fetchExchange, useQuery } from 'urql';
-
-const TODOS_QUERY = `
-  query { todos { id text } }
-`;
+const ssr = ssrExchange();
+const client = createClient({
+  url: 'https://trygql.formidable.dev/graphql/web-collections',
+  exchanges: [cacheExchange, ssr, fetchExchange],
```

Comment:
Since this is using 'use client', what's the best way to get a reference to the context like `withUrqlClient` provides? This is for cases where a need for a different method of passing the Authorization header is required during SSR.

Reply:
That's a good question, `withUrqlClient` obtains this only when it's called through `getInitialProps` next does not expose it elsewhere (apart from `next/headers`) so this might be a question for their side 😅

Reply:
true, I hope they have a proper answer because this is really puzzling, even just passing credentials for an API on the same domain isn't sending the auth cookies when the query is executed!